### PR TITLE
Create script overhaul

### PR DESCRIPTION
**Pull request by liorama**
Add:
* Update existing offers (only for earn) not just create new ones
* Dry run, process but do not change DB - doesn't apply to spend offers
* Everything configurable via command line arguments. -h/--help looks like this now:
```
usage: create [-h] [-v] [--apps-dir APPS_DIR] [--offers-dir OFFERS_DIR]
              [--app-list APP_LIST] [--no-update] [-d] [-c]


DB initializer script for Apps and Offers. Apps configuration come as JSON
files and offers as CSV files.

Optional arguments:
  -h, --help            Show this help message and exit.
  -v, --version         Show program's version number and exit.
  --apps-dir APPS_DIR   Location (directory) of app config json files
  --offers-dir OFFERS_DIR
                        Location (directory) of offers csv files
  --app-list APP_LIST   Comma separated list of apps (i.e smpl, swel, kik,
                        test, p365...) to have the earn offers added to (ALL,
                        in caps, to add to all apps)
  --no-update           Don't update existing earn offers, only create new
                        ones.
  -d, --dry-run         Process the data but don't touch the DB (doesn't apply to spend offers)
  -c, --create-db       Create tables/schemes if needed.  USUALLY SHOULD BE NOT RUN IN PRODUCTION
```

### CODE REVIEW

**Comment by liorama** [comment]
I still need to do update and dry run for spend...maybe

File: node-console.js [hunk 1]
```diff
@@ -46,3 +46,35 @@ OrderContext = _orders.OrderContext;
 User = (r("models/users")).User;
 AuthToken = (r("models/users")).AuthToken;
 Application = (r("models/applications")).Application;
+AppOffer = r('models/applications').AppOffer;
```
Comment (nitzantomer) [inline node-console.js]: why is this file here and not (in ts) with the rest of them scripts?

File: scripts/src/create.ts [hunk 2]
```diff
@@ -18,17 +18,17 @@ import "./models/orders";
 import "./models/users";
 
 getConfig();
+let config;
```
Comment (nitzantomer) [inline scripts/src/create.ts]: this is confusing, change this to be a specific config as a line above you have a different type of config.
also, give it a type annotation here so it will be clear.

File: scripts/src/create.ts [hunk 3]
```diff
@@ -195,35 +188,121 @@ function getStellarAddresses() {
 	}
 }
 
-initModels(true).then(async () => {
-	const appsDir = process.argv[2];
-	const offersDir = process.argv[3];
+type Config = {
```
Comment (nitzantomer) [inline scripts/src/create.ts]: same thing, be specific of what type of config this is

File: scripts/src/create.ts [hunk 4]
```diff
@@ -195,35 +188,121 @@ function getStellarAddresses() {
...
+		});
+	});
+}
 
-	for (const filename of fs.readdirSync(path(appsDir))) {
-		if (!filename.endsWith(".json")) {
-			console.info(`skipping non json file ${filename}`);
-			continue;
+config = initArgsParser();
+console.log(config);
```
Comment (nitzantomer) [inline scripts/src/create.ts]: is this a debugging thing or you want to output the config all the time?

File: scripts/src/create.ts [hunk 5]
```diff
@@ -39,11 +39,12 @@ async function createApp(appId: string, name: string, jwtPublicKeys: StringMap,
 	if (apiKey) {
 		app.apiKey = apiKey;  // when apiKey given, run-over generated value
 	}
-	await app.save();
+	console.log("creating app: %s (id: %s)", name, appId, dryRun ? "(dry run)" : undefined);
```
[the remark this answers is not in the document]
Comment (liorama) [reply scripts/src/create.ts]: There are benefits to handing arguments over to console, it pretty prints objects and in browser it also does so for DOM elements and supports styling logs. I just do this here for consistency.

File: scripts/src/create.ts [hunk 6]
```diff
@@ -89,38 +91,28 @@ async function parseSpend(data: string[][]) {
...
-			v.get("Description")!,
-			v.get("Image")!,
-			parseInt(v.get("Amount")!, 10),
-			parseInt(v.get("CapTotal")!, 10),
-			parseInt(v.get("CapPerUser")!, 10),
-			v.get("OrderTitle")!,
-			v.get("OrderDescription")!,
-			contentType,
-			poll);
+		const offer = await createEarn(v.get("OfferName")!, STELLAR_ADDRESS || v.get("WalletAddress")!, v.get("Brand")!, v.get("Title")!, v.get("Description")!, v.get("Image")!, parseInt(v.get("Amount")!, 10), parseInt(v.get("CapTotal")!, 10), parseInt(v.get("CapPerUser")!, 10), v.get("OrderTitle")!, v.get("OrderDescription")!, contentType, poll, appList, config);
```
[the remark this answers is not in the document]
Comment (liorama) [reply scripts/src/create.ts]: Pycharm formatted it and I forgot to undo it...

File: node-console.js (hunk 1 above)
Comment (liorama) [reply node-console.js]: This needs to be ready even before you build the files.

File: scripts/src/create.ts (hunk 2 above)
Comment (liorama) [reply scripts/src/create.ts]: Cool

File: scripts/src/create.ts (hunk 4 above)
Comment (liorama) [reply scripts/src/create.ts]: leftover debugging thing

File: scripts/src/create.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (liorama) [reply scripts/src/create.ts]: We can talk about this.

File: node-console.js (hunk 1 above)
Comment (nitzantomer) [reply node-console.js]: why?

Comment (liorama) [reply node-console.js]: mmm...I had a big written explanation about when this needs to be ready but it doesn't matter cause it has TS dependencies...I'll convert it

File: .travis.yml [hunk 7]
```diff
@@ -19,6 +19,3 @@ script:
 
 after_failure:
   - make logs
-
-after_script:
-  - make down
```
[the remark this answers is not in the document]
Comment (liorama) [reply .travis.yml]: @doodyparizada Tests pass without this so leaving it this way (sans `make down`)

File: node-console.js (hunk 1 above)
Comment (liorama) [reply node-console.js]: Since this file wasn't added in this PR but only had some minor changes & this PR has been waiting for a long time, I'll convert this in a dedicated PR.
